pkg/aggregator/aggregator.go

```diff
@@ -384,6 +398,30 @@ func (agg *BufferedAggregator) handleSenderBucket(checkBucket senderHistogramBuc
 	}
 }
 
+func (agg *BufferedAggregator) handleEventPlatformEvent(event senderEventPlatformEvent) error {
+	agg.mu.Lock()
+	defer agg.mu.Unlock()
+	if agg.eventPlatformForwarder == nil {
+		return errors.New("event platform forwarder not initialized")
+	}
+	m := &message.Message{Content: []byte(event.rawEvent)}
+	err := agg.eventPlatformForwarder.SendEventPlatformEvent(m, event.eventType)
+	if err != nil {
+		return err
+	}
+	if agg.flushInterval == 0 {
+		// If flushing is disabled then we want to buffer the events in memory to allow them to be read back out
+		// by the check command. During normal operation there is no buffering done in the aggregator for these events.
+		// We only buffer events that were successfully sent to the eventPlatformForwarder in order to aid local
+		// debugging in case of immediate errors, like sending an unsupported eventType.
+		agg.eventPlatformEvents = append(agg.eventPlatformEvents, EventPlatformDebugEvent{
+			RawEvent:  event.rawEvent,
+			EventType: event.eventType,
+		})
+	}
```

Comment:
What happend when in the `eventPlatformForwarder` when ran in the check command ? Do we drop the message ? Queue them ?

I feel this section of code should be done by the `eventPlatformForwarder` and not in the aggregator.

Reply:
Agreed, I've updated it here https://github.com/DataDog/datadog-agent/pull/7735/commits/ad255a6028a6a03c2e9637a5ee2efa3b166cf2c4 

What do you think of that change? 

Previously the check command also forwarding the events to the intake, which didn't match the behavior for the rest of the aggregator, which was not forwarding any other telemetry. Now for the check command we're using a "noop" `EventPlatformForwarder` which buffers events without sending, and the aggregator reads them out.